Improving the version selector in version details

Taj010 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
The version selector in version details is improved. The changes include: 
- Moving the version dropdown next to the model name in the page title
- Removing the helper text in the dropdown and adding a divider there
- A latest tag for the most recently created version in the dropdown 
- "view all versions" link that redirects to the versions page of the model
- Have the header display the Model name instead of the version name

The current UI looks like this: 
<img width="506" height="309" alt="image" src="https://github.com/user-attachments/assets/48091a2f-8999-43c5-9889-a77377471856" />

<!--- [UI] Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
Manually tested in the browser: verified the version selector appears next to the model name, the dropdown expands correctly, the "Latest" label displays, and the "View all versions" link navigates to the correct page.
Also checked that all UI changes render correctly across different models and versions.
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] The commits have meaningful messages
- [ ] Automated tests are provided as part of the PR for major new functionalities; testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).
- [ ] **For first time contributors**: Please reach out to the [Reviewers](https://github.com/kubeflow/model-registry/blob/main/OWNERS) to ensure all tests are being run, ensuring the label `ok-to-test` has been added to the PR.

If you have UI changes

<!--- You can ignore these if you are doing Go Model Registry REST server, MR Python client, manifest, controller, internal logic, etc changes; aka non-UI / visual changes -->
- [ ] The developer has added tests or explained why testing cannot be added.
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

manaswinidas commented on clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelVersionDetails.cy.ts:
```diff
@@ -209,7 +209,7 @@ describe('Model version details', () => {
         '/model-registry/modelregistry-sample/registeredModels/1/versions/1/details',
       );
-      cy.findByTestId('app-page-title').should('have.text', 'Version 1');
+      cy.findByTestId('app-page-title').should('contain.text', 'Version 1');
```
I think we should check for the model name here as well

manaswinidas commented on clients/ui/frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelVersions.cy.ts:
```diff
@@ -251,7 +251,7 @@ describe('Model Versions', () => {
     modelVersionRow.findModelVersionName().contains('model version').click();
     verifyRelativeURL('/model-registry/modelregistry-sample/registeredModels/1/versions/1/details');
-    cy.findByTestId('app-page-title').should('have.text', 'model version');
+    cy.findByTestId('app-page-title').should('contain.text', 'test');
```
add another line for checking for the model name along with the model version dropdown

rsun19 commented on clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionSelector.tsx:
```diff
@@ -104,7 +133,7 @@ const ModelVersionSelector: React.FC<ModelVersionSelectorProps> = ({
       menu={menu}
       menuRef={menuRef}
-      popperProps={{ maxWidth: 'trigger' }}
+      popperProps={{ minWidth: '250px', maxWidth: 'none' }}
```
I wonder if this prop is necessary, since removing it made no difference on my screen. However, let me know if it does.

```suggestion
```

Taj010 replied:
@rsun19 you are right removing it does not make any difference.  But having the initial   `popperProps={{ maxWidth: 'trigger' }} ` part was affecting the dropdown layout but right now without calling the popperProps itself it looks good.

DaoDaoNoCode commented on clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionSelector.tsx:
```diff
@@ -32,15 +39,25 @@ const ModelVersionSelector: React.FC<ModelVersionSelectorProps> = ({
-        {mv.name}
+        <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+          <FlexItem>{mv.name}</FlexItem>
+          <FlexItem>{mv.id === latestVersion.id && <Label color="blue">Latest</Label>}</FlexItem>
```
What's more, we need to add this badge to line 131 in the toggle if this is the latest version in the selector.

Taj010 commented on clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionDetails.tsx:
```diff
@@ -72,28 +80,28 @@ const ModelVersionsDetails: React.FC<ModelVersionsDetailProps> = ({ tab, ...page
+      headerAction={
+        mvLoaded && mv && <ModelVersionsDetailsHeaderActions mv={mv} refresh={refresh} />
       }
       description={<Truncate content={mv?.description || ''} />}
```
@DaoDaoNoCode This was addressed in the other PR - https://github.com/kubeflow/model-registry/pull/1396-

Taj010 commented:
I have addressed the comments and made the changes, some major changes is the placeholder in the search field and changing  the label to badge and added it to the toggle too. Also added test to check for the latest badge in the dropdown and moved the ViewAllVersionsButton to the shared components. 

The current UI looks like:
<img width="832" height="610" alt="image" src="https://github.com/user-attachments/assets/bfffbc2b-d328-49f2-807e-aab96ced0b2e" />

manaswinidas commented on clients/ui/frontend/src/app/shared/components/ViewAllVersionsButton.tsx:
```diff
@@ -0,0 +1,46 @@
+          : modelVersionListUrl(rmId, preferredModelRegistry?.name)
+      }
+      variant="link"
+      icon={<ArrowRightIcon />}
```
We don't need this icon

rsun19 replied:
I see. In this case, to make it reusable, we should make it an optional parameter.

Taj010 commented:
To summarize the changes made through all the comments:

- Added a placeholder to the search field and updated the label to “Badge”; also applied the badge label to the toggle component.
- Refactored code, removed custom CSS and renamed elements accordingly.
- Moved the "View all x versions" link into a shared component; removed the right arrow icon from version selection (is available as an optional parameter).
- Added tests to verify the presence of the "Latest" badge in the dropdown and ensure the "View all x versions" link redirects to the correct versions page and the check for model name

The current UI looks like:
<img width="668" height="392" alt="image" src="https://github.com/user-attachments/assets/9e14fe16-3740-4216-b711-0cc590657a17" />

ppadti commented on clients/ui/frontend/src/app/shared/components/ViewAllVersionsButton.tsx:
```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
```
I saw this component here in [this PR](https://github.com/kubeflow/model-registry/pull/1392/files#diff-83224ec9a2ae0876ba1539fbae77c258eac6e329c3cb49c0af9ce0cd687eabd5) as well. not sure which one will go in first. Will have merge conflict, I guess. cc @rsun19 
may be we can move this component to `/screens/components`
As we will move the files under shared folder to shared library.

rsun19 replied:
Yes I agree lets move it

mturley commented:
The replacement PR https://github.com/kubeflow/model-registry/pull/1443 is now merged, we can close this one (I don't have permission though)

pboyd commented:
At @mturley's request:

/close

mturley commented:
Thanks @pboyd !
